db/migrate/20241129052526_add_unique_index_to_collaborators.rb

```diff
@@ -0,0 +1,18 @@
+class AddUniqueIndexToCollaborators < ActiveRecord::Migration[7.0]
+  def change
+    duplicates = Collaborator.group(:user_id, :todo_list_id)
+    .having('COUNT(*) > 1')
+    .pluck(:user_id, :todo_list_id)
+
+  duplicates.each do |user_id, todo_list_id|
+  # Keep only the first record and delete the rest
+  Collaborator.where(user_id: user_id, todo_list_id: todo_list_id)
+  .order(:id)
+  .offset(1) # Skip the first record
+  .destroy_all
+  end
```

Comment:
Why do we need this in migration?

Reply:
This migration file is designed to add a unique index to the collaborators table for the combination of user_id and todo_list_id.
so basically it help in prevention of duplication of collaborations.